doc/api/zlib.md

```diff
@@ -4,60 +4,101 @@
 
 > Stability: 2 - Stable
 
-The `zlib` module provides compression functionality implemented using Gzip and
-Deflate/Inflate, as well as Brotli. It can be accessed using:
+The `zlib` module provides compression functionality implemented using Gzip,
+Deflate/Inflate, and Brotli.
+
+It can be accessed using:
 
 ```js
 const zlib = require('zlib');
 ```
 
+Compression and decompression is built around the Node.js [Streams API][].
+
 Compressing or decompressing a stream (such as a file) can be accomplished by
-piping the source stream data through a `zlib` stream into a destination stream:
+piping the source stream through a `zlib` `Transform` stream into a destination
+stream:
 
 ```js
-const gzip = zlib.createGzip();
-const fs = require('fs');
-const inp = fs.createReadStream('input.txt');
-const out = fs.createWriteStream('input.txt.gz');
-
-inp.pipe(gzip)
-  .on('error', () => {
-    // handle error
-  })
-  .pipe(out)
-  .on('error', () => {
-    // handle error
+const { createGzip } = require('zlib');
+const { pipeline } = require('stream');
+const {
+  createReadStream,
+  createWriteStream
+} = require('fs');
+
+const gzip = createGzip();
+const source = createReadStream('input.txt');
+const destination = createWriteStream('input.txt.gz');
+
+pipeline(source, gzip, destination, (err) => {
+  if (err) {
+    console.error('An error occurred:', err);
+    process.exit(1);
```

Comment:
Why forcing the process to exit in examples?

Reply:
To emphasize that the error needs to be handled somehow, and typically crashing is safer than continuing unless things are handled properly.

Reply:
Wouldn't this example exit anyway if an error occurs? If so wouldn't it be better to use `process.exitCode = 1` instead of forcing exit if the example wants to set the exit code to 1?

Reply:
That assumption may not hold if the sample code is cut and pasted into some other code.

Reply:
Hmm it doesn't make much sense. What makes you think that the error should be handled by forcing the process to exit if I copy paste that example in my code?

Reply:
I don't care that much just trying to understand what was the reason to add it.

Reply:
There are many ways it could be handled, yes. This being the most draconian which will at least ensure that a dev and copies and pastes the code will go looking for the crash. Open to other recommendations here.
// cc @mcollina

Reply:
Setting `process.exitCode` is better here, imo – or just re-throwing `err`, tbh.

Reply:
Ok, updated to `process.exitCode`